### ISSUE TITLE
Implement _VariableFunctionsClass.randint of torch

### ISSUE THREAD

**Issue by athitten** [Lightning-AI/lightning-thunder#269]
## 🚀 Feature

Implement [`_VariableFunctionsClass.randint`](https://pytorch.org/docs/stable/generated/torch.randint.html)

### Motivation

NeMo [Stable Diffusion](https://github.com/NVIDIA/NeMo/blob/main/examples/multimodal/text_to_image/stable_diffusion/sd_train.py)

**Comment by tfogal** [comment]
triage team: what guarantees do we need to provide w.r.t. the generated random numbers? i.e. do we need to match torch exactly, match the same distribution, merely respect the `low`/`high` parameters?

NeMo *does* occasionally use the `dtype` parameter, and sometimes it's even `float32` or similar instead of some integer. We might want to map this an RNG op + a cast.

**Comment by mruberry** [comment]
triage review — we currently don't require matching anything; we would accept an implementation of randint that calls PyTorch's randint

**Comment by aditya-167** [comment]
Hi.. I would like to work on this.. is the work still in progress?

**Comment by mruberry** [comment]
> Hi.. I would like to work on this.. is the work still in progress?

Yep! No one is currently working on this. It'd be great to see a PR

**Comment by lantiga** [comment]
go for it @aditya-167!
